### PR TITLE
connection_* functions are impure

### PR DESCRIPTION
**Pull request by staabm**
~~requires https://github.com/JetBrains/phpstorm-stubs/pull/1560~~

### CODE REVIEW

File: bin/generate-function-metadata.php [hunk 1]
```diff
@@ -60,6 +61,21 @@ public function enterNode(Node $node)
 			return null;
 		}
 
+		private function isHasSideEffectsAnnotated(Attribute $attr)
```
Comment (staabm) [inline bin/generate-function-metadata.php]: according to a conversation with a contributor on JetBrains/phpstorm-stubs, the `Pure` attribute has [different semantics based on the 1st parameter](https://github.com/JetBrains/phpstorm-stubs/pull/1560#issuecomment-1684166592)

Comment (staabm) [reply bin/generate-function-metadata.php]: the regenerated function map changed a lot.. I am not 100% sure all these changes make sense - at least no tests fail

Comment (staabm) [reply bin/generate-function-metadata.php]: hmm I think I over-think this thing and I should instead just have added a new entry to the `functionMetadata_original`.

**Comment by ondrejmirtes** [comment]
Thank you.

**Pull request merged**